src/main/java/com/redhat/devtools/lsp4ij/settings/ui/JsonTextField.java

```diff
@@ -57,6 +65,25 @@ public JsonTextField(@NotNull Project project) {
         add(editorTextField, BorderLayout.CENTER);
     }
 
+    /**
+     * Sets the editor's filename so that the correct JSON schema will be used for code completion and validation.
+     *
+     * @param jsonFilename the JSON file name
+     */
+    public void setJsonFilename(@NotNull String jsonFilename) {
```

Comment:
This allows us to make a `JsonTextField` use one of the registered JSON schema file providers based on the associated file name.